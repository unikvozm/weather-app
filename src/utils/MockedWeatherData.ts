export const data = {
    cnt: 3,
    list: [
        {
            "coord": {
                "lon": -0.13,
                "lat": 51.51
            },
            "sys": {
                "country": "GB",
                "timezone": 3600,
                "sunrise": 1597812785,
                "sunset": 1597864524
            },
            "weather": [
                {
                    "id": 804,
                    "main": "Clouds",
                    "description": "overcast clouds",
                    "icon": "04d"
                }
            ],
            "main": {
                "temp": 19.27,
                "feels_like": 18.87,
                "temp_min": 18.89,
                "temp_max": 20,
                "pressure": 1001,
                "humidity": 88
            },
            "visibility": 9000,
            "wind": {
                "speed": 4.1,
                "deg": 170
            },
            "clouds": {
                "all": 100
            },
            "dt": 1597857081,
            "id": 2643743,
            "name": "London"
        },
        {
            "coord": {
                "lon": 17.03,
                "lat": 51.1
            },
            "sys": {
                "country": "PL",
                "timezone": 7200,
                "sunrise": 1597808724,
                "sunset": 1597860349
            },
            "weather": [
                {
                    "id": 802,
                    "main": "Clouds",
                    "description": "scattered clouds",
                    "icon": "03d"
                }
            ],
            "main": {
                "temp": 22.19,
                "feels_like": 22.8,
                "temp_min": 21.67,
                "temp_max": 22.78,
                "pressure": 1010,
                "humidity": 73
            },
            "visibility": 10000,
            "wind": {
                "speed": 2.6,
                "deg": 310
            },
            "clouds": {
                "all": 40
            },
            "dt": 1597857172,
            "id": 3081368,
            "name": "Wroc≈Çaw"
        },
        {
            "coord": {
                "lon": 10.45,
                "lat": 54.03
            },
            "sys": {
                "country": "DE",
                "timezone": 7200,
                "sunrise": 1597809831,
                "sunset": 1597862400
            },
            "weather": [
                {
                    "id": 803,
                    "main": "Clouds",
                    "description": "broken clouds",
                    "icon": "04d"
                }
            ],
            "main": {
                "temp": 24.42,
                "feels_like": 23.8,
                "temp_min": 23.33,
                "temp_max": 25,
                "pressure": 1009,
                "humidity": 44
            },
            "visibility": 10000,
            "wind": {
                "speed": 1.5,
                "deg": 170
            },
            "clouds": {
                "all": 63
            },
            "dt": 1597857137,
            "id": 2950158,
            "name": "Berlin"
        }
    ]
}