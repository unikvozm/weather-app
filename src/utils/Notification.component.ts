// packages used to request Native Modules from native side
import {NativeModules} from 'react-native';

class OurModuleInterface {
  static showFancyNotification(title: string, body: string) {
    const module = NativeModules.MyNotification;

    module.showNotification(title, body);
  }
}

export default OurModuleInterface;
