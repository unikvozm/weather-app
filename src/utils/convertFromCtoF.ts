export const convertFromCtoF = (celsius: number): number =>
  Math.round((celsius * 9) / 5 + 32);
