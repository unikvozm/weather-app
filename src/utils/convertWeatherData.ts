import {WeatherDataFetched, WeatherData} from '../types/WeatherData';

export const converWeatherData = (data: WeatherDataFetched): WeatherData => ({
  id: data.id,
  name: data.name,
  icon: data.weather[0].icon,
  description: data.weather[0].description,
  weather: data.weather[0].main,
  temp: data.main.temp,
  pressure: data.main.pressure,
  humidity: data.main.humidity,
  windSpeed: data.wind.speed,
  cloudsCover: data.clouds.all,
  visibility: data.visibility,
});
