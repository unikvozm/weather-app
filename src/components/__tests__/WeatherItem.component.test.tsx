import React from 'react';
import renderer from 'react-test-renderer';
import {WeatherItem} from '../WeatherItem.component';
import {data} from '../../utils/MockedWeatherData';
import {converWeatherData} from '../../utils/convertWeatherData';

const convertedData = converWeatherData(data.list[0]);

test('WeatherItem renders correctly', () => {
  const tree = renderer
    .create(<WeatherItem city={convertedData} units="metric" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
