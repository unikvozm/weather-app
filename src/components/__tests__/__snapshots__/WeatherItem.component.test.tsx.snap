// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherItem renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    accessibilityHint="Weather Forecast"
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "padding": 8,
          },
          Object {
            "borderBottomColor": "#ABACAB",
            "borderBottomWidth": 2,
            "justifyContent": "space-between",
            "margin": 2,
            "padding": 12,
          },
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        accessibilityIgnoresInvertColors={true}
        color="rgba(0, 0, 0, 0.54)"
        source={
          Object {
            "uri": "https://openweathermap.org/img/wn/04d@4x.png",
          }
        }
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
                undefined,
              ],
            ]
          }
        >
          London
        </Text>
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 14,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                },
                undefined,
              ],
            ]
          }
        >
          Clouds
        </Text>
      </View>
      <View
        color="rgba(0, 0, 0, 0.54)"
        style={
          Object {
            "backgroundColor": "#3EEFD8",
            "borderColor": "#3EEFD8",
            "borderRadius": 16,
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "elevation": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
              "selected": false,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 16,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 4,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "marginVertical": 4,
                      "minHeight": 24,
                      "textAlignVertical": "center",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "marginLeft": 8,
                      "marginRight": 8,
                    },
                    Object {
                      "alignItems": "center",
                      "fontSize": 20,
                      "padding": 10,
                    },
                  ],
                ]
              }
            >
              19.27Â°C
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
